src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,46 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { User } from '../../types/User';
+import { getTodos, getUser } from '../../api';
+import { Todo } from '../../types/Todo';
+
+export interface TodoModalProps {
+  selected: number;
+  onSelected: (id: number) => void;
+}
+
+export const TodoModal: React.FC<TodoModalProps> = ({
+  selected,
+  onSelected,
+}) => {
+  const [user, setUser] = useState<User>();
+  const [toDoList, setToDoList] = useState<Todo[]>([]);
+  const [isLoadingTodos, setIsLoadingTodos] = useState(true);
```

Comment:
looks too complex, let's simplify it

Reply:
we can save here the whole todo in the state 
<img width="505" alt="image" src="https://github.com/user-attachments/assets/ad6fe4d0-16fe-4038-af05-ff272971bcbb">


then you will pass that selectedTodo into the modal
<img width="591" alt="image" src="https://github.com/user-attachments/assets/97c2ff44-7d6e-4c6c-a491-953bc51e7749">


and just call the getUser function